components/common/src/main/java/com/hotels/styx/server/HttpInterceptorContext.java

```diff
@@ -17,18 +17,59 @@
 
 import com.hotels.styx.api.HttpInterceptor;
 
+import java.net.InetSocketAddress;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * ConcurrentHashMap backed implementation of HttpInterceptor.Context.
  */
 public final class HttpInterceptorContext implements HttpInterceptor.Context {
     private final Map<String, Object> context = new ConcurrentHashMap<>();
-    private boolean secure;
 
-    private HttpInterceptorContext(boolean secure) {
+    private final boolean secure;
+    private final InetSocketAddress clientAddress;
+
+    /**
+     * Construct a new instance.
+     *
+     * @param secure true if the request was received via SSL
+     * @param clientAddress address that request came from, or null if not-applicable
+     */
+    public HttpInterceptorContext(boolean secure, InetSocketAddress clientAddress) {
         this.secure = secure;
+        this.clientAddress = clientAddress; // intentionally nullable
+    }
+
+    /**
+     * Construct a new instance.
+     *
+     * @param clientAddress address that request came from, or null if not-applicable
+     */
+    public HttpInterceptorContext(InetSocketAddress clientAddress) {
+        this(false, clientAddress);
+    }
+
+    /**
+     * Construct a new instance.
+     *
+     * @param secure true if the request was received via SSL
+     */
+    public HttpInterceptorContext(boolean secure) {
+        this(secure, null);
+    }
+
+    /**
+     * Construct a new instance.
+     */
+    public HttpInterceptorContext() {
+        this(false, null);
+    }
+
+    // TODO deprecate
+    public static HttpInterceptor.Context create() {
```

Comment:
BTW why are you so keen to deprecate this?

Reply:
The static-factory-method isn't really adding anything. It seems to just be a totally redundant wrapper for the constructor.

Sometimes when we add static-factory-methods to our classes it's to make it read more fluently when statically-imported, but this is just called `create` so it doesn't have that benefit.

It just changes `new HttpInterceptorContext()` to `HttpInterceptorContext.create()` which is actually longer and less convenient.